src/main/java/build/buildfarm/worker/Worker.java

```diff
@@ -70,17 +72,30 @@ private static Path getValidCasCacheDirectory(WorkerConfig config, Path root) th
     return root.resolve(casCacheValue);
   }
 
+  private static HashFunction getValidHashFunction(WorkerConfig config) throws ConfigurationException {
+    switch (config.getHashFunction()) {
```

Comment:
One way to not duplicate this would be to add a cast constructor to the DigestUtil.HashFunction enum that takes this proto enum.

Reply:
Done, though with a static getter, since we already constructed them...